### DISCUSSION TITLE
Cannot commit - File too big

### DISCUSSION THREAD

**Discussion by coeur-de-loup** [Vinzent03/obsidian-git#732] [Problems]
I like to put my my project files aside my obsidian files, in the same folder. Would it be possible to setup obsidian git with git LFS, so the big files can also be backuped? Right now I get "Cannot commit - File is too big"

**Comment by Vinzent03** [comment]
You should be able to set up git lfs normally and the plugin just works. If not, there are already some issues which may help you.

**Reply by TimonGisler** [reply]
Hi, I recently set up LFS for .aup3 files.

But for some reason, the plugin still gives me the error "Did not commit, because following files are too big: fileName."
If I do it manually in the windows CMD terminal, it works perfectly fine.

I tried:

1. Removing the file from the cache (git rm --cached "file")
2. Reinstalling the plugin
3. Restarting computer/Obsidian
4. Setting custom git directory path
5. Cloning my notes into a new folder and creating a new vault
6. setting git config --local core.bigFileThreshold 2g
7. Added the output of where git-lfs to both the PATH variable of my system and to the "Additional PATH environment variable paths" setting of the plugin
8. Cloned the repo to another laptop
9. git config lfs.enabled true

None of it worked...
I went through "https://github.com/Vinzent03/obsidian-git/issues?q=lfs" and "https://github.com/Vinzent03/obsidian-git/discussions?discussions_q=lfs," but none of it seemed relevant or helped.

Is there, by any chance, a known bug or workaround?
What seems weird to me, the error suggests that it fails when it tries to commit, but commiting should never be a problem, pushing may be a problem, so it confuses me a little bit that it seems to fail on commit already.

I can do it manually (git add. and git commit -m "smth" and git push), but it is a little bit annoying.
Any help is appreciated, this is driving me insane : )

**Reply by TimonGisler** [reply]
I did some very very surface level digging, could it be that the plugin checks the file size before commiting anything?

More specifically could this code be the culprit?:
```
    async function commit() {
        loading = true;
        if (status) {
            if (await plugin.tools.hasTooBigFiles(status.staged)) { 
            ....
```


Edit: Update, i just created an PR which should fix this problem: https://github.com/Vinzent03/obsidian-git/pull/822
